```
fcitx5-chinese-addons> -- Detecting CXX compile features - done
fcitx5-chinese-addons> CMake Error at CMakeLists.txt:4 (find_package):
fcitx5-chinese-addons>   By not providing "FindECM.cmake" in CMAKE_MODULE_PATH this project has
fcitx5-chinese-addons>   asked CMake to find a package configuration file provided by "ECM", but
fcitx5-chinese-addons>   CMake did not find one.
fcitx5-chinese-addons>
fcitx5-chinese-addons>   Could not find a package configuration file provided by "ECM" with any of
fcitx5-chinese-addons>   the following names:
fcitx5-chinese-addons>
fcitx5-chinese-addons>     ECMConfig.cmake
fcitx5-chinese-addons>     ecm-config.cmake
fcitx5-chinese-addons>
fcitx5-chinese-addons>   Add the installation prefix of "ECM" to CMAKE_PREFIX_PATH or set "ECM_DIR"
fcitx5-chinese-addons>   to a directory containing one of the above files.  If "ECM" provides a
fcitx5-chinese-addons>   separate development package or SDK, be sure it has been installed.
fcitx5-chinese-addons>
fcitx5-chinese-addons>
fcitx5-chinese-addons> -- Configuring incomplete, errors occurred!
```

--- a/pkgs/tools/inputmethods/fcitx5/fcitx5-lua.nix
+++ b/pkgs/tools/inputmethods/fcitx5/fcitx5-lua.nix
@@ -22,12 +22,12 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [
     cmake
     extra-cmake-modules
+    gettext
   ];
 
   buildInputs = [
     fcitx5
     lua
-    gettext
   ];
 
   passthru = {

--- a/pkgs/tools/inputmethods/fcitx5/fcitx5-table-extra.nix
+++ b/pkgs/tools/inputmethods/fcitx5/fcitx5-table-extra.nix
@@ -26,8 +26,11 @@ stdenv.mkDerivation rec {
     extra-cmake-modules
     gettext
     libime
-    boost
     fcitx5
+  ];
+
+  buildInputs = [
+    boost
   ];
 
   meta = with lib; {

--- a/pkgs/tools/inputmethods/fcitx5/fcitx5-table-other.nix
+++ b/pkgs/tools/inputmethods/fcitx5/fcitx5-table-other.nix
@@ -26,8 +26,11 @@ stdenv.mkDerivation rec {
     extra-cmake-modules
     gettext
     libime
-    boost
     fcitx5
+  ];
+
+  buildInputs = [
+    boost
   ];
 
   meta = with lib; {

